packages/storybook-readme/src/services/getPropsTables/refinePropTables.js

```diff
@@ -0,0 +1,45 @@
+// Refine user inputs(excludePropTables, includePropTables) by type guarding
+// whether it is an array and a function. For now, type checking if it
+// is a real React component(class, stateless, renderable) looks overkill
+// since displayName and name is only be used.
+export const refinePropTables = (excludePropTables, includePropTables) => {
+  if (!Array.isArray(excludePropTables) || !Array.isArray(includePropTables)) {
+    console.warn(
```

Comment:
Maybe convert to array automatically and check if all elements are valid?

Reply:
Hm... what you are saying is if inputs are not an array, then make an array and put that value into it and filter its values? I think this logic covers this case that users do something like this
```
// It still works because I refine this value to array with the passed component
includePropTables: ReactComponent
```
I think It depends on the user experience... either
1. Trying as much as possible to make things work even if users pass bad inputs
2. If bad inputs are passed just ignore everything.
Of course both with warning messages.

Current implementation is somewhere between 1 and 2
1. If both inputs are not an array, it is really bad. I just ignore all this rules by returning empty array.(with warning messages)
2. If array is passed, but its values are not React component, then it is less bad? I just filter only React components and apply this rules.(with warning messages)

My other solution was to just ignore all this rules if 
1. both inputs are not an array
2. and its values are not a React component
What is good about this is that it is either working or not. It doesn't filter valid values like current implementation which may confuses users that sometimes it works and sometimes it doesn't. 

#### If you want it like first case(trying refine user inputs as much as possible)
I will add what you said(whatever inputs `string, boolean, React component` are first put into an array and filter its values)

#### If you prefer, I would say "NO bad inputs ever!"
I will change it to my other solution(ignore all if inputs are bad)

Reply:
Completely agree with second solution

Reply:
I applied second solution, please have a look